ext/standard/array.c

```diff
@@ -4122,14 +4122,15 @@ zend_bool array_column_param_helper(zval *param,
 
 		case IS_OBJECT:
 			if (!try_convert_to_string(param)) {
+				zend_type_error("Cannot convert key %s to a string", name);
```

Comment:
try_convert_to_string will throw itself if it fails, so this will result in a duplicate exception.